flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/writer/RecordWriterTest.java

```diff
@@ -412,23 +412,24 @@ public void testBroadcastEmitRecord() throws Exception {
 	 *
 	 * @param isBroadcastEmit whether using {@link RecordWriter#broadcastEmit(IOReadableWritable)} or not
 	 */
+	@SuppressWarnings("unchecked")
 	private void emitRecordWithBroadcastPartitionerOrBroadcastEmitRecord(boolean isBroadcastEmit) throws Exception {
 		final int numChannels = 4;
 		final int bufferSize = 32;
 		final int numValues = 8;
 		final int serializationLength = 4;
 
-		@SuppressWarnings("unchecked")
 		final Queue<BufferConsumer>[] queues = new Queue[numChannels];
 		for (int i = 0; i < numChannels; i++) {
 			queues[i] = new ArrayDeque<>();
 		}
 
 		final TestPooledBufferProvider bufferProvider = new TestPooledBufferProvider(Integer.MAX_VALUE, bufferSize);
 		final ResultPartitionWriter partitionWriter = new CollectingPartitionWriter(queues, bufferProvider);
+		final ChannelSelector selector = new Broadcast<>();
+		selector.setup(numChannels);
```

Comment:
Won't this be called inside `RecordWriter`'s constructor?

Reply:
Yeah, it is useless code. I would remove it.